pkgs/tools/networking/curl/netrc-regression.patch

```diff
@@ -0,0 +1,16 @@
+diff --git a/lib/url.c b/lib/url.c
+index a56e4b0..9f29593 100644
+--- a/lib/url.c
++++ b/lib/url.c
+@@ -2971,6 +2971,12 @@ static CURLcode override_login(struct Curl_easy *data,
+       /* don't update the user name below */
+       userp = NULL;
+     }
++    /* no user was set but a password, set a blank user */
++    if(userp && !*userp && *passwdp) {
```

Comment:
If I read this correctly, a blank user will also be set if the password is blank. That's not necessary for cachix and it's slightly risky.
Does curl come with a test that exercises the usual user+pass both set combination, to make sure we don't accidentally clear the user? And do we run it?

Reply:
It comes with a test suite, but it doesn't seem to have caught this bug. And yes, we run the tests.

About the empty password case, do you have a suggestion how to handle it better?

Reply:
I've misread it, but let's be safe and check the pointer before checking for a blank string.
```suggestion
+    if(userp && !*userp && passwdp && *passwdp) {
```

ie make the blank check null-pointer safe.

Honestly C code with this much pointers makes me very anxious. I have a lot of respect for people who can program in such an environment.